### PR TITLE
[DOCS] RHEL Security TMP override

### PR DESCRIPTION
**Pull request by stefnestor**
Adding content for [top last 30d Elastic Discuss post](https://discuss.elastic.co/t/elasticsearch-v7-6-2-failed-to-start-killed-by-sigabrt-on-rhel-7-7-urgent/231039) which covers a BlackHat RHEL start-up error with [this solution](https://discuss.elastic.co/t/elasticsearch-v7-6-2-failed-to-start-killed-by-sigabrt-on-rhel-7-7-urgent/231039/11).

Not sure if it makes sense to include "if you don't do this, you'll get [these errors](https://discuss.elastic.co/t/elasticsearch-v7-6-2-failed-to-start-killed-by-sigabrt-on-rhel-7-7-urgent/231039/8)" or some kind of better framing/context. Putting in raw helpful content for review 🙏🏼

### CODE REVIEW

File: docs/reference/setup/install/rpm.asciidoc [hunk 1]
```diff
@@ -123,6 +123,17 @@ endif::[]
 
 include::skip-set-kernel-parameters.asciidoc[]
 
+[[install-rhel-security]]
+==== Override Red Hat TMP if Security Compliance
```
Comment (pugnascotia) [inline docs/reference/setup/install/rpm.asciidoc:127]: Should be something like the following? This feels convoluted.

```suggestion
==== Override Red Hat TMP if Security Compliant
```

File: docs/reference/setup/install/rpm.asciidoc [hunk 2]
```diff
@@ -123,6 +123,17 @@ endif::[]
...
+
+If you are applying a security compliance in your RHEL installation 
+you must change the path of the TMP directory that will use elasticsearch as Java.
```
Comment (pugnascotia) [inline docs/reference/setup/install/rpm.asciidoc:130]:
```suggestion
you must change the path of the TMP directory that Java and {es} will use.
```

File: docs/reference/setup/install/rpm.asciidoc [hunk 3]
```diff
@@ -123,6 +123,17 @@ endif::[]
...
+
+1. Uncomment `-Djava.io.tmpdir=${ES_TMPDIR}` under `/etc/elasticsearch/jvm.options`
+2. Add `ES_TMPDIR=/usr/share/elasticsearch/tmp` under `/etc/sysconfig/elasticsearch`
+3. Create the /usr/share/elasticsearch/tmp directory and make sure that the owner and group are elasticsearch and the permissions are `0755`
```
Comment (pugnascotia) [inline docs/reference/setup/install/rpm.asciidoc:134]:
```suggestion
3. Create the `/usr/share/elasticsearch/tmp` directory and make sure that the owner and group are `elasticsearch` and the permissions are `0755`
```

**Comment by pugnascotia** [comment]
@DaveCTurner knows more about this than me, so I've added him as a reviewer.

File: docs/reference/setup/install/rpm.asciidoc [hunk 4]
```diff
@@ -123,6 +123,17 @@ endif::[]
 
 include::skip-set-kernel-parameters.asciidoc[]
 
+[[install-rhel-security]]
+==== Override Red Hat TMP if Security Compliance
+
+If you are applying a security compliance in your RHEL installation 
+you must change the path of the TMP directory that will use elasticsearch as Java.
+
+1. Uncomment `-Djava.io.tmpdir=${ES_TMPDIR}` under `/etc/elasticsearch/jvm.options`
```
Comment (DaveCTurner) [inline docs/reference/setup/install/rpm.asciidoc:132]: I think we shouldn't be telling users to edit `jvm.options`, they should be adding extra options files in `jvm.options.d`.

But then I think this isn't necessary anyway, we already copy `ES_TMPDIR` to `java.io.tmpdir` IIRC.

File: docs/reference/setup/install/rpm.asciidoc [hunk 5]
```diff
@@ -123,6 +123,17 @@ endif::[]
...
+2. Add `ES_TMPDIR=/usr/share/elasticsearch/tmp` under `/etc/sysconfig/elasticsearch`
+3. Create the /usr/share/elasticsearch/tmp directory and make sure that the owner and group are elasticsearch and the permissions are `0755`
+4. Lastly make sure that `/dev/shm` doesn't have the noexec attribute with command: `mount | grep tmpfs | grep '/dev/shm'`
```
Comment (DaveCTurner) [inline docs/reference/setup/install/rpm.asciidoc:135]: This is probably unacceptable to many users of locked-down systems. Instead we should set `TMPDIR` (eventually `LIBFFI_TMPDIR` to point to a location to which only the `elasticsearch` user has access.

**Review by DaveCTurner — COMMENTED** [review]
I think we shouldn't copy the instructions from the forum post without validating them and considering some simpler alternatives. I believe we could fix this whole thing ourselves by setting `TMPDIR` to the value of `ES_TMPDIR` in our startup scripts. Does anything bad happen if we set `TMPDIR` for every Linux user?

A more targeted fix will be https://github.com/elastic/elasticsearch/issues/77014 but this needs JNA to upgrade their `libffi` and then we need to upgrade our JNA so it'll be a while before we can do that.

I'm also not sure we actually need to override `ES_TMPDIR` in these environments either. If you hit the segfault in the linked forum post then we know that JNA's temporary directory supports executables so it should be enough to tell `libffi` to use the same location.

**Comment by DaveCTurner** [comment]
#74163 is related but kinda different, `libffi` doesn't care about system properties so fixing the temp dir for `libffi` is fairly orthogonal to overriding `java.io.tmpdir`. Not that I hold a strong opinion about how we organise these issues, up to you and the core/infra folks really.

**Review by debadair — COMMENTED** [review]
I added some comments and suggestions. It took me a bit to wrap my head around what was going on here, so not 100% sure the details are correct. However, I do thing we need to work in a bit more context about what you're doing and why. 

Also, we should add an IMPORTANT note to https://www.elastic.co/guide/en/elasticsearch/reference/master/important-settings.html#es-tmpdir that indicates that you must use a dedicated TMP directory outside of the system TMP directory if you are running on a system with FIPS mode enabled.

**Comment by DaveCTurner** [comment]
Now that #80651 is merged I think users don't need to do most of this stuff. It should be sufficient to just set `ES_TMPDIR` to point somewhere other than a subdirectory of `/tmp`. If we want to write step-by-step instructions for that then IMO it'd be better in `docs/reference/setup/sysconfig/executable-jna-tmpdir.asciidoc` since they are applicable to all distributions, not just RPM. Admittedly it affects RHEL folks more than others, so a cross-link from these RPM docs would be helpful.

**Comment by stefnestor** [comment]
@debadair I'm suppose to  ... but I don't show the option from the documentation ... . Can you help me convert to whatever Docs needs to move forward? 🙏🏼 

> @stefnestor please enable the option "Allow edits and access to secrets by maintainers" on your PR. For more information, [see the documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

<details><summary>screenshot</summary>

image

</details>

**Comment by pugnascotia** [comment]
Gah. The GitHub API isn't very helpful when it comes to detecting the "Allow edits and access to secrets by maintainers" setting, sorry for the noise

**Comment by stefnestor** [comment]
Deb's no longer with Elastic ( cc: @elastic/docs ) & per @DaveCTurner  ... above, I'll side with this no longer being needed & close. Cheers all!

> Now that https://github.com/elastic/elasticsearch/pull/80651 is merged I think users don't need to do most of this stuff.

**Pull request closed without merging**